torch/csrc/jit/export.cpp

```diff
@@ -235,9 +245,33 @@ void encodeGraph(onnx::GraphProto * p_g, const std::shared_ptr<Graph> & g,
     for(auto attr_name : node->attributeNames()) {
       addAttribute(p_n, node, attr_name);
     }
+    if (node->kind() == torch::jit::onnx::Loop) {
+      JIT_ASSERT(node->blocks().size() == 1);
```

Comment:
I wasn't paying attention when support for blocks was added to the core IR, so this comment is a bit late... but, I am curious why blocks aren't implemented using the graph attribute in attributes? In that case, shouldn't this be a no-op (assuming that graph attribute export is done correctly.)

Reply:
@ezyang I think it's because attribute graphs can't implicitly capture values from outer blocks, while our block mechanism allows them to use `Value`s from dominating blocks.

Reply:
OK, so, does the conversion from block to graph undo the captures? Since I don't see it in this patch, I guess this code must have already been implemented somewhere?

Reply:
Since all value references in ONNX are done using identifier strings rather than direct pointers, we don't actually need them to have explicit captures, we can just allow the free variables to exist in the protobufs. I think the control-flow proposal in ONNX also does it this way, so we don't need capture logic.